discojs/src/aggregator/aggregator.ts

```diff
@@ -62,24 +62,53 @@ export abstract class Base<T> extends EventEmitter<{'aggregation': T }> {
     this.contributions = Map()
     this._nodes = Set()
 
-    // On every aggregation, update the object's state to match the current aggregation
-    // and communication rounds.
-    this.on('aggregation', () => this.nextRound())
+    // On each aggregation, increment
+    // updates the aggregator's state to proceed to the next communication round.
+    // If all communication rounds were performed, proceeds to the next aggregation round
+    // and empties the collection of stored contributions.
+    this.on('aggregation', () => {
+      if (++this._communicationRound === this.communicationRounds) {
```

Comment:
mixing test and actions is error prone (and makes my brain melt)

```suggestion
      this._communicationRound++;
      if (this.communicationRound === this.communicationRounds) {
```